lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -446,14 +454,17 @@ def autoscale(self, enable=True, axis='both', tight=None) :
             scaley = False
             scalez = False
             if axis in ['x', 'both']:
-                self._autoscaleXon = bool(enable)
-                scalex = self._autoscaleXon
+                self._autoscaleXon = scalex = bool(enable)
+            else:
+                scalex = False
             if axis in ['y', 'both']:
-                self._autoscaleYon = bool(enable)
-                scaley = self._autoscaleYon
+                self._autoscaleYon = scaley = bool(enable)
+            else:
+                scaly = False
```

Comment:
typo, plus, the else-clause isn't needed as the scale{x,y,z} variables are already initialized to False